src/bufr/BufrParser/Exports/Variables/TimeoffsetVariable.cpp

```diff
@@ -0,0 +1,168 @@
+ */
+
+#include <iomanip>
+#include <iostream>
+#include <locale>
+#include <ostream>
+#include <ctime>
+#include <vector>
+#include <unordered_map>
+
+#include "eckit/exception/Exceptions.h"
+#include "oops/util/Logger.h"
+
+#include "DataObject.h"
+#include "TimeoffsetVariable.h"
+#include "Transforms/TransformBuilder.h"
+
+
+namespace
+{
+    namespace ConfKeys
+    {
+        const char* Timeoffset = "time_offset";
+        const char* Referencetime = "reference_time";
+        const char* GroupByField = "group_by";
+        const char* Transforms = "transforms";
+    }  // namespace ConfKeys
+}  // namespace
+
+
+namespace Ingester
+{
+    TimeoffsetVariable::TimeoffsetVariable(const std::string& exportName,
+                                           const std::string& groupByField,
+                                           const eckit::LocalConfiguration &conf) :
+      Variable(exportName),
+      conf_(conf)
+    {
+        initQueryMap();
+    }
+
+    std::shared_ptr<DataObjectBase> TimeoffsetVariable::exportData(const BufrDataMap& map)
+    {
+        checkKeys(map);
+
+        std::tm tm{};                // zero initialise
+        tm.tm_year = 1970 - 1900;    // 1970
+        tm.tm_mon = 0;               // Jan=0, Feb=1, ...
+        tm.tm_mday = 1;              // 1st
+        tm.tm_hour = 0;              // midnight
+        tm.tm_min = 0;
+        tm.tm_sec = 0;
+        tm.tm_isdst = 0;             // Not daylight saving
+        std::time_t epochDt = std::mktime(&tm);
+
+        // Convert the reference time (ISO8601 string) to time struct
+        std::tm ref_time = {};
+        std::istringstream ss(conf_.getString(ConfKeys::Referencetime));
+
+#ifndef __APPLE__
+        // ss.imbue(std::locale("en_US.utf-8"));
+#endif
+
+        ss >> std::get_time(&ref_time, "%Y-%m-%dT%H:%M:%S");
+        if (ss.fail())
+        {
+            std::ostringstream errStr;
+            errStr << "Reference time MUST be formatted like 2021-11-29T22:43:51Z";
+            throw eckit::BadParameter(errStr.str());
+        }
+
+        auto timeOffsets = map.at(getExportKey(ConfKeys::Timeoffset));
+        if (conf_.has(ConfKeys::Transforms))
+        {
+            auto transforms = TransformBuilder::makeTransforms(conf_);
+            for (const auto &transform : transforms)
+            {
+                transform->apply(timeOffsets);
+            }
+        }
+
+        auto timeDiffs = std::vector<int64_t> (timeOffsets->size());
+        for (size_t idx = 0; idx < timeOffsets->size(); ++idx)
+        {
+            auto diff_time = DataObject<int64_t>::missingValue();
+            if (!timeOffsets->isMissing(idx))
+            {
+                auto obs_tm = ref_time;
+                obs_tm.tm_sec = ref_time.tm_sec + timeOffsets->getAsInt(idx);
+                auto thisTime = std::mktime(&obs_tm);
+                diff_time = static_cast<int64_t>(difftime(thisTime, epochDt));
```

Comment:
Local variables should be camel case starting with lower case letter. For example **diff_time** should be **diffTime**.

Reply:
If it is OK with you, I wish to leave this alone, because it is the same variable name as done before in the datetime code.  So for consistency, can we just leave this one be?

Reply:
ok